### DISCUSSION TITLE
Optix 7 and Yocto-gl

### DISCUSSION THREAD

**Discussion by Hurleyworks** [xelatihy/yocto-gl#1293] [Show and tell]
Howdy,

I managed to cobble together a path tracer using NVIDIA's Optix7 and some of Yocto-gl's trace functions converted to cuda. I am only able to get the "Naive path tracer"  working so far:
https://www.youtube.com/watch?v=_q9e39FAoJU&ab_channel=Hurleyworks

Optix7 has a pretty steep learning curve but Shocker-0x15 has kindly provided an excellent wrapper that takes a lot of pain out of getting up and running
https://github.com/shocker-0x15/OptiX_Utility

Here's the Yocto-gl trace functions that I used converted to cuda 
https://gist.github.com/Hurleyworks/6fb983964304496d3b9eee800e639fd4

Thanks for making such an amazingly well done library available!

**Comment by xelatihy** [comment]
Wow! This is very nice!

I wanted to try an OptiX port, but I cannot find the time for it just yet.
This is a great first start.

As for the shaders, the naive integrator should be easier to port, while the complete one should be much harder since it requires ray tracing inside the shader for the PDF. One possibility is to support only light sources with very few triangles in the real-time one. But a better solution is to eventually make a light accelerator that takes care of this. Probably won't get done for a while though, until I can fin a way to keep the library small.

**Comment by Hurleyworks** [comment]
I haven't been able to make much progress on the complete path tracer but I've been having fun with instancing using the naive one using just image based lighting.

http://berserko.hurleyworks.com/portfolio-items/cg/#
